### PR TITLE
Intelligent dataset features

### PR DESCRIPTION
**Pull request by prescod**
Datasets are intelligently defaulted so that they should seldom "error out". It uses CCI code which knows how to find an extractable dataset.

When you select a field, if the field is a Lookup, the Lookup target is auto-selected.

### CODE REVIEW

**Review by boakley — COMMENTED** [review]
There are a small handful of print, pprint, and console.log statements. Were those intentional side-effects of the functions they are in, or were they accidentally left in as a debugging tool?

File: metecho/api/jobs.py [hunk 1]
```diff
@@ -1150,13 +1158,39 @@ def user_reassign(scratch_org, *, new_user, originating_user_id):
 
 user_reassign_job = job(user_reassign)
 
+# TODO: Move some of this into CCI itself
+def get_objs_and_fields_from_org(schema: Schema) -> Dict[str, Dict]:
+    extract_file = StringIO(DEFAULT_EXTRACT_DATA)
+    decls = ExtractRulesFile.parse_extract(extract_file)
+    flattened = flatten_declarations(list(decls.values()), schema)
+    print("F", [f.sf_object for f in flattened])
```
[the remark this answers is not in the document]
Comment (prescod) [reply metecho/api/jobs.py]: Yeah I noticed and cleaned those up a few hours ago but your browser window might have been stale.

9c251ee

[the remark this answers is not in the document]
Comment (prescod) [reply metecho/api/jobs.py]: I finally got all linting to pass, tests to pass and 100% coverage.

**Review by boakley — APPROVED** [review]
This is very unfamiliar code to me, but I don't see any red flags now that there is good test coverage and the print statements have been removed.

**Pull request merged**